packages/storybook/babel.config.js

```diff
@@ -0,0 +1,63 @@
+const path = require('path');
+
+const errorCodesPath = path.resolve(__dirname, './public/static/error-codes.json');
+
+let defaultPresets;
+
+// We release a ES version of Material-UI.
+// It's something that matches the latest official supported features of JavaScript.
+// Nothing more (stage-1, etc), nothing less (require, etc).
+if (process.env.BABEL_ENV === 'es') {
+  defaultPresets = [];
+} else {
+  defaultPresets = [
+    [
+      '@babel/preset-env',
+      {
+        bugfixes: true,
+        modules: ['esm', 'production-umd'].includes(process.env.BABEL_ENV) ? false : 'commonjs',
+      },
+    ],
+  ];
+}
+
+const alias = {
+  '@material-ui/x-grid': '../../packages/grid/x-grid/src',
```

Comment:
I'd like to see if it works with just a webpack alias

Reply:
I think it can work with webpack aliases only yet, 90% chance. The motivation for using babel is to apply the same code transformation optimization we would use with the component we publish.

Reply:
A thought on the bundling strategy. On the core, we have 5M downloads/month, for material-ui-x, we can look at https://npm-stat.com/charts.html?package=ag-grid-enterprise to get a reasonable expectation of the usage. I think that, in the best case, we can hope for 1/10 fewer downloads (but x100 more $/download). In this context, it can be wiser to use the strategy of the core as more stress-tested, we will get more feedback, and hence, are likely to deliver the best tradeoff.

Reply:
I think it's a VERY radical solution to sort a simple problem. 
It also seems that we want to bring a bunch of tools blindly without assessing the necessity of each tool.
The test infrastructure seems overly complex and not in the scope of the issue of the storybook build.

What is babel going to bring? We use typescript, which does everything we need to compile and check our code.

Also I believe babel is just a transpiler hence, the following code will get transpiled without error with babel and throw an error with typescript.
`const a: string = 10 `;

Reply:
> The test infrastructure seems overly complex and not in the scope of the issue of the storybook build.

The issue description and git commit history should help. There are 3 commits, only the last one is meant to solve the problem. 

But mind that the pull request is in a draft mode as waiting on the other pull request, it's based on, to be resolved. Sorry for the confusion. 

Regarding the complexity of it, I think that we should focus on the output of it, we will bundle it so it can be shared without leading to duplication between the two repositories, also so you don't have to know about the work required to make it run correctly :). 

> I think it's a VERY radical solution to sort a simple problem.

I'm fine with using webpack only. The only change required I'm aware of is that we would need to replace the absolute import in the storybook configure file to import the license release generator from it's relative location. It's not a big deal.

> It also seems that we want to bring a bunch of tools blindly without assessing the necessity of each tool.

It depends on what you mean by "bringing", we are already using these tools. I think that we should treat this repository as if we would merge it with the main one tomorrow (while we won't), same objective: build leading, in their scope, React components. 

> What is babel going to bring? We use typescript, which does everything we need to compile and check our code.

We leverage Babel, on top of TypeScript to apply optimizations on the code, before we release it on npm or before we push it to a CDN. It's not something TypeScript alone supports. 

You are right, Babel is no replacement for TypeScript, both tools are complementary. This make me think that we miss the building step in the CI, something to assert that:

- There is no TypeScript error
- We can build and publish the packages
- The published packages work

One step at the time, will come later :)